## <center>Практическое задание к лекции 1.

**1) Из колоды в 52 карты извлекаются случайным образом 4 карты.**   
   **a) Найти вероятность того, что все карты – крести.**  
   **б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.**  
   
---

a) В колоде из 52 листов $\frac{52}{4}=13$ карт одной масти.  
Воспользуемся формулой подсчета количества сочетаний по $k$ элементов из множества $n$:  
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$  
И классической формулой вероятности:  
$\displaystyle P(A)=\frac{m}{n}$ где $m$-число **благоприятных** исходов, $n$-общее число **всех** исходов.  
в нашем случае $\displaystyle n=C^4_{52}$-т.е. количество способов извлечь 4 карты из 52, а $\displaystyle m=C^4_{13}$, способов извлечь 4 карты одной масти.  
Для облегчения расчетов напишем функцию:

In [1]:
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k))) 

Вычислим $\displaystyle m$:

In [2]:
m=combinations(13, 4)
print(f'm = {m}')

m = 715


Вычислим $\displaystyle n$:

In [4]:
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


Найдем вероятность $P(4 крести)$

In [5]:
P=m/n
print(f'P(4 крести) = {round(P,4)}')

P(4 трефы) = 0.0026


----
б) в этом случае так же общее число исходов $\displaystyle n=C^4_{52}$  
а $\displaystyle m$ будет :  
    из 4-х карт 0 тузов:  
   число способов достать 0 тузов из 4-х $C^4_{48}$
   
   
Вычислим:  

In [7]:
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


In [9]:
m=combinations(48, 4)
print(f'm = {m}')

m = 194580


Найдем вероятность $P(\text{хотя бы 1 туз, это 1-вероятность не достать ни одного})$

In [10]:
P=1-m/n
print(f'P(хотя бы 1 туз) = {round(P,4)}')

P(хотя бы 1 туз) = 0.2813


---  

#### Ответ:  
а) $P(\text{4 крести})\approx 0.0026 \approx 0.26\%$;  
б) $P(\text{хотя бы 1 туз})\approx 0.2813 \approx 28.13\%$.  

---

**2) На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?**  

---

Так как кнопки нажимаются **одновременно**, то порядок не важен, и для формулы вероятности $\displaystyle P(A)=\frac{m}{n}$ общее число исходов будет определяться количиством сочетаний 3-х элементов из 10, $\displaystyle n=C^3_{10}$, а количество благоприятных исходов $\displaystyle m=1$-замок открылся.
Вычислим $\displaystyle P(A)$:  

In [19]:
P=1/combinations(10,3)
print(f'P(открыть с первой попытки) = {round(P,4)}')

P(открыть с первой попытки) = 0.0083


---

#### Ответ:   
$\displaystyle P(\text{открыть с первой попытки})\approx0.0083\approx0.83\%$  

---

**3) В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?**  

---

Общее количество исходов будет определяться количеством сочетаний 3 деталей из 15 $\displaystyle n=C^3_{15}$, а количество благоприятных исходов - количеством сочетаний 3-х окрашенных деталей из 9  $\displaystyle m=C^3_{9}$  
Вычислим $P(\text{3 из 3-х окрашены})$:

In [20]:
P=combinations(9,3)/combinations(15,3)
print(f'P(3 из 3-х окрашены) = {round(P,4)}')

P(3 из 3-х окрашены) = 0.1846


---

#### Ответ:  
$P(\text{3 из 3-х окрашены})\approx0.1846\approx18.46\%$  

---

**4) В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?**  

---

$\displaystyle P(\text{оба выигрышных})=\frac{1}{C^2_{100}}=\frac{1}{\frac{100!}{2!(100-2)!}}=\frac{2!\cdot98!}{100!}$ или $\displaystyle P(\text{оба выигрышных})=\frac{1}{100}\cdot\frac{1}{99}$  
Вычислим:

In [22]:
P=1/combinations(100,2)
print(f'P(оба выигрышных = {round(P,4)})')

P(оба выигрышных = 0.0002)


---

#### Ответ:  
$\displaystyle P(\text{оба выигрышных})\approx0.0002\approx0.02\%$  

---